test/renamer_test.dart

```diff
@@ -0,0 +1,174 @@
+// https://opensource.org/licenses/MIT.
+
+import 'package:sass_migrator/src/renamer.dart';
+import 'package:test/test.dart';
+
+void main() {
+  group('single statements', () {
+    group('simple rename with', () {
+      test('sed-style syntax', () {
+        var renamer = Renamer.simple('s/old/new/');
+        expect(renamer.rename('old'), equals('new'));
+        expect(renamer.rename('oldx'), isNull);
+        expect(renamer.rename('xold'), isNull);
+        expect(renamer.rename('new'), isNull);
+      });
+
+      test('default key', () {
+        var renamer = Renamer.simple('/old/new/');
+        expect(renamer.rename('old'), equals('new'));
+        expect(renamer.rename('oldx'), isNull);
+        expect(renamer.rename('xold'), isNull);
+        expect(renamer.rename('new'), isNull);
+      });
+
+      test('space syntax with `to` clause', () {
+        var renamer = Renamer.simple('s old to new');
+        expect(renamer.rename('old'), equals('new'));
+        expect(renamer.rename('oldx'), isNull);
+        expect(renamer.rename('xold'), isNull);
+        expect(renamer.rename('new'), isNull);
+      });
+
+      test('space syntax with `to` clause and default key', () {
+        var renamer = Renamer.simple('old to new');
+        expect(renamer.rename('old'), equals('new'));
+        expect(renamer.rename('oldx'), isNull);
+        expect(renamer.rename('xold'), isNull);
+        expect(renamer.rename('new'), isNull);
+      });
+
+      test('space syntax with `->` clause', () {
+        var renamer = Renamer.simple('s old -> new');
+        expect(renamer.rename('old'), equals('new'));
+        expect(renamer.rename('oldx'), isNull);
+        expect(renamer.rename('xold'), isNull);
+        expect(renamer.rename('new'), isNull);
+      });
+
+      test('space syntax with `->` clause and default key', () {
+        var renamer = Renamer.simple('old -> new');
+        expect(renamer.rename('old'), equals('new'));
+        expect(renamer.rename('oldx'), isNull);
+        expect(renamer.rename('xold'), isNull);
+        expect(renamer.rename('new'), isNull);
+      });
+    });
+
+    group('backreferences', () {
+      test(r'to entire match', () {
+        var renamer = Renamer.simple(r'.+ to prefix-\0');
+        expect(renamer.rename('a'), equals('prefix-a'));
+        expect(renamer.rename('abc'), equals('prefix-abc'));
+      });
+
+      test(r'to group', () {
+        var renamer = Renamer.simple(r'/(.+)-suffix/\1/');
+        expect(renamer.rename('a-suffix'), equals('a'));
+        expect(renamer.rename('abc-suffix'), equals('abc'));
+        expect(renamer.rename('abc'), isNull);
+        expect(renamer.rename('-suffix'), isNull);
+      });
+    });
+
+    group('escaping', () {
+      test(r'spaces', () {
+        var renamer = Renamer.simple(r'a\ b to x\ y');
+        expect(renamer.rename('a b'), equals('x y'));
+      });
+
+      test(r'normal delimiters', () {
```

Comment:
Also test a delimiter at the beginning of the source pattern and escaped backslashes, including things like `foo to bar\\;` and `foo to bar\\\;`.